dap_prinz_green_jobs/pipeline/green_measures/industries/industries_measures.py

```diff
@@ -0,0 +1,193 @@
+        self.ghg_emissions_dict, self.ghg_unit_emissions_dict = get_clean_ghg_data()
+        # Dictionary of SIC sector (e.g. "A") to proportion of hours worked spent doing green tasks
+        self.sic_section_2_prop_hours = create_section_dict(
+            load_green_tasks_prop_hours()
+        )
+        # Dictionary of SIC sector (e.g. "A") to proportion of workers doing green tasks
+        self.sic_section_2_prop_workers = create_section_dict(
+            load_green_tasks_prop_workers()
+        )
+        # Dictionary of SIC sector (e.g. "A") to proportion of workers spending at least 20% of
+        # their time doing green tasks per SIC
+        self.sic_section_2_prop_workers_20 = create_section_dict(
+            load_green_tasks_prop_workers_20()
+        )
+
+        sic_data = load_sic()
+        self.sic_names = dict(
+            zip(sic_data["Most disaggregated level"], sic_data["Description"])
+        )
+        self.sic_names = {str(k).strip(): v for k, v in self.sic_names.items()}
+
+        self.sic_to_section = {
+            k: v.strip()
+            for k, v in dict(
+                zip(sic_data["Most disaggregated level"], sic_data["SECTION"])
+            ).items()
+        }
+
+    def get_measures(
+        self, job_adverts: Union[Dict[str, str], List[Dict[str, str]]]
+    ) -> List[Dict[str, float]]:
+        """Extract industry-level green measures for a given job advert
+            or list of job adverts.
+
+        Args:
+            job_adverts Union[Dict[str, str], List[Dict[str, str]]]: A job advert
+                as a dictionary or list of dictionaries.
+
+        Returns:
+            Dict[str, float]: Industry-level green measures
+                for a given job advert or list of job adverts.
+        """
+        sic_codes = self.sm.get_sic_codes(job_adverts)
+
+        industry_measures_list = []
+        for sic_info in sic_codes:
+            confidence_threshold = None
+            if sic_info["sic_method"] == "closest distance":
+                confidence_threshold = self.closest_distance_threshold
+            elif sic_info["sic_method"] == "majority":
+                confidence_threshold = self.majority_sic_threshold
+
+            if (
+                confidence_threshold is not None
+                and sic_info["sic_confidence"] > confidence_threshold
+            ):
+                sic = sic_info["sic_code"]
+            elif sic_info["sic_method"] == "companies house":
+                sic = sic_info["sic_code"]
+            else:
+                sic = None
+
+            sic_clean = clean_sic(sic)
+            sic_section = self.sic_to_section.get(sic_clean)
+            industry_measures = {
+                "SIC": sic,
+                "SIC_name": self.sic_names.get(sic),
+                "SIC_confidence": sic_info["sic_confidence"],
+                "SIC_method": sic_info["sic_method"],
```

Comment:
I tried applying this to 100 job adverts and I found outputs where SIC was None, but the confidence and method were given. e.g.: 

>>> [(s['SIC'], s['SIC_confidence'], s['SIC_method']) for s in green_industry_outputs_dict.values() if not s['SIC']]
[(None, 0.32, 'majority SIC'), (None, None, None), (None, None, None), (None, 0.49, 'closest distance'), (None, 0.34, 'majority SIC'), (None, None, None), (None, 0.28, 'majority SIC'), (None, None, None), (None, 0.5, 'closest distance'), (None, 0.5, 'closest distance'), (None, 0.3, 'majority SIC'), (None, None, None), (None, 0.3, 'majority SIC'), (None, 0.21, 'majority SIC'), (None, 0.24, 'majority SIC'), (None, 0.49, 'closest distance'), (None, None, None), (None, None, None), (None, 0.32, 'majority SIC'), (None, 0.32, 'majority SIC'), (None, 0.27, 'majority SIC'), (None, 0.49, 'closest distance'), (None, 0.31, 'majority SIC'), (None, 0.29, 'majority SIC'), (None, 0.26, 'majority SIC'), (None, 0.29, 'majority SIC'), (None, None, None), (None, 0.32, 'majority SIC'), (None, 0.32, 'majority SIC'), (None, None, None), (None, 0.33, 'majority SIC'), (None, 0.3, 'majority SIC'), (None, None, None), (None, 0.48, 'closest distance'), (None, 0.34, 'majority SIC'), (None, None, None), (None, 0.36, 'majority SIC')]

So if SIC==None perhaps don't output the confidence and the method?

I'm also not sure why there are some outputs like `(None, 0.34, 'majority SIC')`? how did the confidence score get so high if the SIC code is None?

Reply:
I wonder if there's a small bug in industries measures - my thought was the not do a ton of thresholding in the SicMapper and do more of that in the industries measures class - i'll take a look